README.md

```diff
@@ -15,27 +15,33 @@
 </a>
 
 # Generative data preparation
-This software package is designed for preparing data that can be used to train generative models. It offers an efficient way to convert input text files into tokenized sequences that are packed into a fixed sequence length. The resulting output directory can be directly used for training with SambaStudio. This package features many styles of packing text of any length into tokenized sequences, compressed HDF5 file outputs, efficient multiprocessing, shuffling any sized dataset, splitting your data into train/dev/test, and specifying what tokens are attended to during training.
+
+The [`pipeline.py`](https://github.com/sambanova/generative_data_prep/blob/main/generative_data_prep/data_prep/pipeline.py) script streamlines data preparation for machine learning model training. It takes a single input file, shuffles and splits it into train/dev/test files, and tokenizes, sequences, and converts them to HDF5 format using
+the utilities in [`data_prep.py`](https://github.com/sambanova/generative_data_prep/blob/main/generative_data_prep/data_prep/data_prep.py). The output directory contains multiple split HDF5 files that are needed to run data parallel training. This output directory will be directly used as a training dataset in SambaStudio. While this package features simple flows that work out of the box, it also supports more customization allowing for many styles of packing text of any length into tokenized sequences.
+
+</br>
 
 ## Table of contents
```

Comment:
Here's some of my rationale for the changes in order here:

1. Left #Requirements and #Installation as the first things the user sees since they are the first chronological steps for a new user.

1. Left `#Requirements` and `#Installation` as the first things the user sees since they are the first chronological steps for a new user.
2. `#Get Started` is the first thing the user sees after installing. I've seen a large body of users wanting to learn as they go instead of reading the whole doc first. This section gets them started with the bare bones requirements and explains the important parameters. 
3. If the user chooses to read the whole document, I thought the ordering should be consistent with the ordering of the ML Pipeline, so for data, I started with `#Input` and `#Output`
4. Next `#Flags` where they can perform any custom tweaks to their data prep process
5. I put `#Examples` towards the bottom since they are better understood after the user has read the previous sections. I put it before `#Understanding Command Outputs` however because I thought it makes sense for the user to see examples of the flags they read about in the section before in action, before making a slight context switch to understand the outputs.
6. Lastly `#Understanding Command Outputs` came last since it's the last part of the process in the data prep journey